password_security/controllers/main.py

```diff
@@ -44,6 +45,7 @@ def web_login(self, *args, **kw):
             request.params['password']
         )
         if not uid:
+            request.uid = old_uid
```

Comment:
Oh I see, because the old/new UID trick is still in place as well. Ok so I checked out your branch and played - you're totally right. We can emulate the sudo using a similar trick as you, but just slightly more explicit:


```
from odoo import http, SUPERUSER_ID
...
    @http.route()
    def web_login(self, *args, **kw):
        ensure_db()
        response = super(PasswordSecurityHome, self).web_login(*args, **kw)
        if not request.httprequest.method == 'POST':
            return response
        uid = request.session.authenticate(
            request.session.db,
            request.params['login'],
            request.params['password']
        )
        if not uid:
            request.uid = SUPERUSER_ID
            request.uid = request.env.ref('base.public_user').id
            return response
        users_obj = request.env['res.users'].sudo()
        user_id = users_obj.browse(request.uid)
        if not user_id._password_has_expired():
            return response
        user_id.action_expire_password()
        redirect = user_id.partner_id.signup_url
        return http.redirect_with_hash(redirect)
```